src/key/key.js

```diff
@@ -413,7 +418,7 @@ Key.prototype.validate = async function() {
   const signatureType = enums.signature.binary;
   signature.signatureType = signatureType;
   await signature.sign(signingKeyPacket, data);
-  return signature.verify(signingKeyPacket, signatureType, data);
+  await signature.verify(signingKeyPacket, signatureType, data);
```

Comment:
on line 417 `hashAlgorithm` is set statically. Should it be checked if it is safe one? It seems that better way to set it would be in config.

Reply:
We could use `config.prefer_hash_algorithm` there, yeah. But that signature is just temporary, we don't export it, and we should eventually check that the key parameters correspond directly without making a signature there.